opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -349,6 +362,316 @@ def collect(
+                mapping = LogarithmMapping(new_scale)
+
+            self._mapping = mapping
+
+            index = self._mapping.map_to_index(value)
+
+            if (
+                index < buckets._index_start
+                and (buckets._index_end - index) >= self._max_size
+            ) or (
+                index > buckets._index_end
+                and (index - buckets._index_start) >= self._max_size
+            ):
+                raise Exception("Downscaling logic error")
+
+        if index < buckets._index_start:
+            span = buckets._index_end - index
+
+            if span >= len(buckets.counts):
+                buckets.grow(span + 1, self._max_size)
+
+            buckets._index_start = index
+
+        elif index > buckets._index_end:
+            span = index - buckets._index_start
+
+            if span >= len(buckets.counts):
+                buckets.grow(span + 1, self._max_size)
+
+            buckets._index_end = index
+
+        bucket_index = index - buckets._index_base
+
+        if bucket_index < 0:
+            bucket_index += len(buckets.counts)
+
+        buckets.increment_bucket(bucket_index)
+
+    def collect(
+        self,
+        aggregation_temporality: AggregationTemporality,
+        collection_start_nano: int,
+    ) -> Optional[_DataPointVarT]:
+        """
+        Atomically return a point for the current value of the metric.
+        """
+
+        with self._lock:
+            if not any(self._negative.counts) and not any(
+                self._positive.counts
+            ):
+                return None
+
+            start_time_unix_nano = self._start_time_unix_nano
+            sum_ = self._sum
+            max_ = self._max
+            min_ = self._min
+
+            self._negative._counts = [0]
+            self._positive._counts = [0]
+            self._start_time_unix_nano = collection_start_nano
+            self._sum = 0
+            self._min = inf
+            self._max = -inf
+
+        if self._count == self._zero_count:
+            scale = 0
+
+        else:
+            scale = self._mapping.scale
+
+        current_point = ExponentialHistogramDataPoint(
+            attributes=self._attributes,
+            start_time_unix_nano=start_time_unix_nano,
+            time_unix_nano=collection_start_nano,
+            count=(
+                sum(self._negative.counts)
+                + sum(self._positive.counts)
+                + self._zero_count
+            ),
+            sum=sum_,
+            scale=scale,
+            zero_count=self._zero_count,
+            positive=BucketsPoint(
+                offset=self._positive.offset,
+                bucket_counts=self._positive.counts,
+            ),
+            negative=BucketsPoint(
+                offset=self._negative.offset,
+                bucket_counts=self._negative.counts,
+            ),
+            # FIXME: Find the right value for flags
+            flags=0,
+            min=min_,
+            max=max_,
```

Comment:
I suppose the exporter has to handle the case where `count_ == 0` meaning that min and max are +/- infinity. As long as the exporter omits these, it's OK. Otherwise, since the protocol has these as optional, should these be None when `count_ == 0`?

Reply:
Right, fixed :v: